File: 1839-longest-substring-of-all-vowels-in-order/1839-longest-substring-of-all-vowels-in-order.js
/**
 * @param {string} word
 * @return {number}
 */
var longestBeautifulSubstring = function(word) {
  const n = word.length;
  let [ans, count, j] = [0, 1, 0];
  
  for (let i = 1; i < n; i++) {
    if (word[i] < word[i - 1]) {
      count = 1;
      j = i;
    }
    else if (word[i] > word[i - 1]) count++;
    if (count === 5) ans = Math.max(ans, i - j + 1);
  }
  
  return ans;
};

File: 1839-longest-substring-of-all-vowels-in-order/1839-longest-substring-of-all-vowels-in-order.py
class Solution:
    def longestBeautifulSubstring(self, word: str) -> int:
        ans, j, count = 0, 0, 1
        for i in range(1, len(word)):
            if word[i] < word[i - 1]:
                count = 1
                j = i
            elif word[i] > word[i - 1]:
                count += 1
            if count == 5:
                ans = max(ans, i - j + 1)
        return ans